app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt

```diff
@@ -94,204 +88,201 @@ class NewMessageFragment : Fragment() {
     }
 
-    fun closeAutocompletion() {
-        fun FieldType.clearField() = getInputView(this).setText("")
-        TO.clearField()
-        CC.clearField()
-        BCC.clearField()
+    private fun setOnKeyboardListener(callback: (isOpened: Boolean) -> Unit) {
+        ViewCompat.setOnApplyWindowInsetsListener(requireActivity().window.decorView) { _, insets ->
+            val isKeyboardVisible = insets.isVisible(WindowInsetsCompat.Type.ime())
+            callback(isKeyboardVisible)
+            insets
+        }
     }
 
-    private fun initDraftAndUi() {
-        newMessageViewModel.initDraftAndUi(newMessageActivityArgs).observe(viewLifecycleOwner) { isSuccess ->
-            if (isSuccess) {
-                observeContacts()
-                populateUiWithExistingDraftData()
-                handleActionSend()
-            } else {
-                requireActivity().finish()
+    private fun observeContacts() {
+        newMessageViewModel.getMergedContacts().observe(viewLifecycleOwner, ::updateContactsAdapter)
+    }
+
+    private fun updateContactsAdapter(allContacts: List<MergedContact>) = with(binding) {
+        toField.updateContacts(allContacts)
+        ccField.updateContacts(allContacts)
+        bccField.updateContacts(allContacts)
+    }
+
+    private fun toggleAutoCompletion(field: FieldType? = null, isAutoCompletionOpened: Boolean) = with(binding) {
+        preFields.isGone = isAutoCompletionOpened
+        to.isVisible = !isAutoCompletionOpened || field == TO
+        cc.isVisible = !isAutoCompletionOpened || field == CC
+        bcc.isVisible = !isAutoCompletionOpened || field == BCC
+        postFields.isGone = isAutoCompletionOpened
+
+        newMessageViewModel.isAutoCompletionOpened = isAutoCompletionOpened
+    }
+
+    private fun populateUiWithViewModel() = with(newMessageViewModel) {
+        attachmentAdapter.addAll(mailAttachments)
+        binding.attachmentsRecyclerView.isGone = attachmentAdapter.itemCount == 0
+        binding.subjectTextField.setText(mailSubject)
+        binding.bodyText.setText(mailBody)
+        mailSignature?.let {
+            binding.signatureWebView.loadDataWithBaseURL("", it, ClipDescription.MIMETYPE_TEXT_HTML, Utils.UTF_8, "")
+            binding.removeSignature.setOnClickListener {
+                mailSignature = null
+                binding.separatedSignature.isGone = true
             }
+            binding.separatedSignature.isVisible = true
         }
+        binding.toField.initRecipients(mailTo)
+        binding.ccField.initRecipients(mailCc)
+        binding.bccField.initRecipients(mailBcc)
     }
 
-    private fun handleActionSend() {
-        when (requireActivity().intent?.action) {
-            Intent.ACTION_SEND -> handleSingleSendIntent()
-            Intent.ACTION_SEND_MULTIPLE -> handleMultipleSendIntent()
-        }
+    private fun populateViewModelWithExternalMailData() {
+        handleMailTo()
+        handleActionSend()
```

Comment:
This way we can simplify it by putting everything together and avoid unnecessary operations.